client/my-sites/sharing/connections/account-dialog.jsx

```diff
@@ -20,24 +20,37 @@ class AccountDialog extends Component {
 		isVisible: PropTypes.bool,
 		onAccountSelected: PropTypes.func,
 		service: PropTypes.object,
+		translate: PropTypes.func,
+		warningNotice: PropTypes.func,
 	};
 
 	static defaultProps = {
 		accounts: Object.freeze( [] ),
 		isVisible: true,
 		onAccountSelected: () => {},
 		service: Object.freeze( {} ),
+		translate: () => {},
+		warningNotice: () => {},
 	};
 
+	onClose = ( action ) => {
+		const accountToConnect = this.getAccountToConnect();
+
+		if ( 'connect' === action && accountToConnect ) {
+			this.props.onAccountSelected( this.props.service, accountToConnect.keyringConnectionId, accountToConnect.ID );
+		} else {
+			this.props.onAccountSelected();
+		}
+	};
+
+	onSelectedAccountChanged = ( account ) => this.setState( { selectedAccount: account } );
+
 	constructor( props ) {
 		super( props );
 
 		this.state = {
 			selectedAccount: null
 		};
```

Comment:
Optional, but we can set default state as a class property initializer, and remove the constructor:
```javascript
state = {
    selectedAccount: null
};
```